common/projectconfig/projectconfig.go

```diff
@@ -55,6 +55,22 @@ func GetDefaultConfigPath() string {
 	return filepath.Join(internal.GitRoot(""), "ftl-project.toml")
 }
 
+// CreateDefaultFileIfNonexistent creates the ftl-project.toml file in the Git root if it
+// does not already exist.
+func CreateDefaultFileIfNonexistent(ctx context.Context) error {
+	path := GetDefaultConfigPath()
```

Comment:
Unfortunately `GitRoot()` can return an empty string, so this can be incorrect. Would you mind converting that function and `GetDefaultConfigPath()` to use `Option[string]` so this is more obvious in the future? You'll need to cater to this here as well.

Reply:
I'm not sure what we should do if we can't find the git root...

Reply:
Ahh, will do. I'll add a warning for that branch case until we come up with something better.

Reply:
Done. Let me know what you think of the behavior for all the other `GitRoot` callers... I defaulted to preserving the old behavior wherever the right choice was ambiguous. At least it's obvious now

Reply:
Another consideration is whether to pass through the original error from `GitRoot` returning `(string, error)`, but I left it as the `optional` for now.